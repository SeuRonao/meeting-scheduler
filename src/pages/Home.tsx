import { useTranslation } from "react-i18next";

export default function Home() {
  const { t } = useTranslation("translation", { keyPrefix: "pages.Home" });
  return (
    <>
      <h1 className="text-center my-3">{t("title")}</h1>
      {t("intro-array", { joinArrays: "\n" })}
      <h2 className="my-3">{t("motivation")}</h2>
      {t("motivation-array", { joinArrays: "\n" })}
      <h2 className="my-3">{t("use")}</h2>
      {t("use-array", { joinArrays: "\n" })}
      <h2 className="my-3">{t("challenges")}</h2>
      {t("challenges-array", { joinArrays: "\n" })}
      <ul>
        {(
          t("challenges-bullet-array", { returnObjects: true }) as Array<string>
        ).map((value, index) => {
          return <li key={index}>{value}</li>;
        })}
      </ul>
      <h2 className="my-3">{t("learning")}</h2>
      {t("learning-array", { joinArrays: "\n" })}
      <ul>
        {(
          t("learning-bullet-array", { returnObjects: true }) as Array<string>
        ).map((value, index) => {
          return <li key={index}>{value}</li>;
        })}
      </ul>
    </>
  );
}
