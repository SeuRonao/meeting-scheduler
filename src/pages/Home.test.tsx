import { create, act, ReactTestRenderer } from "react-test-renderer";
import Home from "./Home";

describe("Testing Profile", () => {
  test("renders correctly", async () => {
    let component: ReactTestRenderer;
    await act(async () => {
      component = create(<Home />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
