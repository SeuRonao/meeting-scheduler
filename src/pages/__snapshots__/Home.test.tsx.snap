// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Profile renders correctly 1`] = `
Array [
  <h1
    className="text-center my-3"
  >
    title
  </h1>,
  "intro-array",
  <h2
    className="my-3"
  >
    motivation
  </h2>,
  "motivation-array",
  <h2
    className="my-3"
  >
    use
  </h2>,
  "use-array",
  <h2
    className="my-3"
  >
    challenges
  </h2>,
  "challenges-array",
  <ul>
    <li>
      challenges-bullet-array
    </li>
    <li>
      challenges-bullet-array
    </li>
  </ul>,
  <h2
    className="my-3"
  >
    learning
  </h2>,
  "learning-array",
  <ul>
    <li>
      learning-bullet-array
    </li>
    <li>
      learning-bullet-array
    </li>
  </ul>,
]
`;
